rclcpp_generic/include/rclcpp_generic/generic_publisher.hpp

```diff
@@ -0,0 +1,91 @@
+// Copyright 2018, Bosch Software Innovations GmbH.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef RCLCPP_GENERIC__GENERIC_PUBLISHER_HPP_
+#define RCLCPP_GENERIC__GENERIC_PUBLISHER_HPP_
+
+#include <memory>
+#include <string>
+
+#include "rclcpp/publisher_base.hpp"
+#include "rclcpp/node_interfaces/node_base_interface.hpp"
+#include "rclcpp/node_interfaces/node_topics_interface.hpp"
+#include "rclcpp/macros.hpp"
+#include "rclcpp/qos.hpp"
+#include "rclcpp/callback_group.hpp"
+#include "rcpputils/shared_library.hpp"
+
+namespace rclcpp_generic
+{
+
+/**
+ * This class is an implementation of an rclcpp::PublisherBase for serialized messages whose topic
+ * is not known at compile time (hence templating does not work).
+ *
+ * It does not support intra-process handling.
```

Comment:
This was copied from `GenericSubscriber` and I'm not actually sure if it's true – what does it mean that intra-process handling is not supported?

Reply:
There is an `IntraProcessManager` in `rclcpp` that circumvents `rcl` in order to implement zero-copy optimization for communication within a single process (potentially multiple nodes, perhaps as Components), see https://github.com/ros2/rclcpp/blob/master/rclcpp/include/rclcpp/experimental/intra_process_manager.hpp

This feature is missing the ability to do some DDS functionality because it is essentially a communication reimplementation (if i understand it correctly) for this limited use case